thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detection/LegacyAnomalyFunctionTranslator.java

```diff
@@ -26,83 +19,44 @@
+  public LegacyAnomalyFunctionTranslator(AnomalyFunctionFactory anomalyFunctionFactory) {
     this.metricConfigDAO = DAO_REGISTRY.getMetricConfigDAO();
     this.anomalyFunctionFactory = anomalyFunctionFactory;
-    this.alertFilterFactory = alertFilterFactory;
   }
 
   /**
    * Translate a legacy anomaly function to a new detection config.
    */
   public DetectionConfigDTO translate(AnomalyFunctionDTO anomalyFunctionDTO) throws Exception {
-    Map<String, Object> properties = new HashMap<>();
-    properties.put(PROP_CLASS_NAME, LegacyMergeWrapper.class);
-    properties.put("anomalyFunctionClassName", anomalyFunctionFactory.getClassNameForFunctionType(anomalyFunctionDTO.getType()));
-
-    String filters = anomalyFunctionDTO.getFilters();
-    MetricConfigDTO metricDTO = this.metricConfigDAO.findByMetricAndDataset(anomalyFunctionDTO.getMetric(), anomalyFunctionDTO.getCollection());
-    if (metricDTO == null) {
-      LOGGER.error("Cannot find metric {} for anomaly function {}", anomalyFunctionDTO.getMetric(), anomalyFunctionDTO.getFunctionName());
-      throw new Exception("Cannot find metric");
-    }
-    anomalyFunctionDTO.setMetricId(metricDTO.getId());
-
-    MetricEntity me = MetricEntity.fromMetric(1.0, metricDTO.getId()).withFilters(ThirdEyeUtils.getFilterSet(filters));
-    String metricUrn = me.getUrn();
-
-    Map<String, Object> legacyAnomalyFunctionProperties = new HashMap<>();
-    legacyAnomalyFunctionProperties.put(PROP_CLASS_NAME, LegacyAnomalyFunctionAlgorithm.class);
-    Map<String, Object> nestedProperties = new HashMap<>();
-
-    if (!Strings.isNullOrEmpty(anomalyFunctionDTO.getExploreDimensions())) {
-      // if anomaly function does dimension exploration, then plug in the dimension wrapper
-      nestedProperties.put(PROP_CLASS_NAME, LegacyAnomalyFunctionAlgorithm.class);
-      nestedProperties.put(PROP_METRIC_URN, metricUrn);
-      nestedProperties.put(PROP_NESTED, Collections.singletonList(legacyAnomalyFunctionProperties));
-      properties.put(PROP_NESTED, Collections.singletonList(nestedProperties));
-    } else {
-      legacyAnomalyFunctionProperties.put(PROP_METRIC_URN, metricUrn);
-      properties.put(PROP_NESTED, Collections.singletonList(legacyAnomalyFunctionProperties));
-    }
-
-    // plug in alert filter if exists
-    Map<String, String> alertFilter = anomalyFunctionDTO.getAlertFilter();
-    if (alertFilter != null) {
-      Map<String, Object> alertFilterProp = new HashMap<>();
-      alertFilterProp.put(PROP_CLASS_NAME, LegacyAlertFilterWrapper.class);
-      String alertFilterType = alertFilter.get("type");
-
-      if (alertFilterType != null) {
-        String alertFilterClassName = this.alertFilterFactory.getClassNameForAlertFilterType(alertFilterType.toUpperCase());
-        if (alertFilterClassName != null) {
-          alertFilterProp.put(PROP_LEGACY_ALERT_FILTER_CLASS_NAME, alertFilterClassName);
```

Comment:
Looks like still need an alert filter class name for the regression alert filter to work.

Reply:
good catch. I re-introduced the logic for this. Also, there's #3149 to add support for global impact filters to the legacy emulation layer